packages/core/src/index.ts

```diff
@@ -240,6 +279,8 @@ export class Abby<
       }
     }
 
+    const storedValue = this.config.flagCacheConfig?.refetchFlags ?  this.getValidFlag(key) : this.#data.flags[key];
```

Comment:
what about doing this:

```suggestion
    const storedValue = this.getValidFlag(key, this.config.flagCacheConfig?.refetchFlags);
```

and adjusting functions arguments..